modules/ocl/feature_match.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * cv_feature_match.h - optical flow feature match
+ *
+ *  Copyright (c) 2016-2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Wind Yuan <feng.yuan@intel.com>
+ * Author: Yinhang Liu <yinhangx.liu@intel.com>
+ */
+
+#ifndef XCAM_FEATURE_MATCH_H
+#define XCAM_FEATURE_MATCH_H
+
+#include <base/xcam_common.h>
+#include <base/xcam_buffer.h>
+#include <dma_video_buffer.h>
+#include <smartptr.h>
+#include "xcam_obj_debug.h"
+#include "image_file_handle.h"
+#include "ocl/cv_base_class.h"
+#include "interface/data_types.h"
+
+#include <ocl/cl_context.h>
+#include <ocl/cl_device.h>
+#include <ocl/cl_memory.h>
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/ocl.hpp>
+
+#define XCAM_CV_FM_MATCH_NUM  2
+
+namespace XCam {
+
+struct CVFMConfig {
+    int sitch_min_width;
+    int min_corners;           // number of minimum efficient corners
+    float offset_factor;       // last_offset * offset_factor + cur_offset * (1.0f - offset_factor)
+    float delta_mean_offset;   // cur_mean_offset - last_mean_offset
+    float max_adjusted_offset; // max offset of each adjustment
+
+    CVFMConfig ()
+        : sitch_min_width (56)
+        , min_corners (8)
+        , offset_factor (0.8f)
+        , delta_mean_offset (5.0f)
+        , max_adjusted_offset (12.0f)
+    {}
+};
+
+class FeatureMatch
+    : public CVBaseClass
```

Comment:
No need to derive from CVBaseClass.
cv_capi_feature_match doesn't need CVBaseClass.
FeatureMatch is only like a interface for other features.
CVFeatureMatch can derive from both FeatureMatch and CVBaseClass.

Reply:
done.